daml-lf/archive/encoder/src/main/scala/com/daml/lf/archive/DarWriter.scala

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf.archive
+
+import java.io.{ByteArrayOutputStream, OutputStream}
+import java.util.zip.{ZipEntry, ZipOutputStream}
+import scalaz.syntax.traverse._
+
+object DarWriter {
+  private val manifestPath = "META-INF/MANIFEST.MF"
+
+  def encode(sdkVersion: String, dar: Dar[(String, Array[Byte])], out: OutputStream): Unit = {
```

Comment:
You could add a comment, stating that out will be closed.

Reply:
Maybe another version with `out: Path` would be handy.

Reply:
good idea on both d21f3ae11c1ad40eff38c757d893209fbed1a375

Reply:
Perfect.